src/pages/actualites/index.page.test.tsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * @jest-environment jsdom
+ */
+import '~/test-utils';
+
+import { render, screen, within } from '@testing-library/react';
+
+import { mockUseRouter } from '~/client/components/useRouter.mock';
+import { mockScrollIntoView, mockSmallScreen } from '~/client/components/window.mock';
+import { DependenciesProvider } from '~/client/context/dependenciesContainer.context';
+import { aManualAnalyticsService } from '~/client/services/analytics/analytics.service.fixture';
+import ActualitesPage, { getStaticProps } from '~/pages/actualites/index.page';
+import { anActualite, anActualiteList } from '~/server/actualites/domain/actualite.fixture';
+import { anArticle } from '~/server/articles/domain/article.fixture';
+import { createFailure, createSuccess } from '~/server/errors/either';
+import { ErreurMetier } from '~/server/errors/erreurMetier.types';
+import { dependencies } from '~/server/start';
+
+
+jest.mock('~/server/start', () => ({
+	dependencies: {
+		actualitesDependencies: {
+			consulterActualitesUseCase: {
+				handle: jest.fn(),
+			},
+		},
+		cmsDependencies: {
+			duréeDeValiditéEnSecondes: jest.fn(),
+		},
+	},
+}));
+
+describe('Page Actualités', () => {
+	beforeEach(() => {
+		mockSmallScreen();
+		mockUseRouter({});
+		mockScrollIntoView();
+	});
+
+	it('lorsque le feature flip ne permet pas l‘accès à la page, redirige vers la page 404', async () => {
+		process.env.NEXT_PUBLIC_OLD_ESPACE_JEUNE_FEATURE = '1';
+
+		const result = await getStaticProps();
+
+		expect(result).toEqual({ notFound: true });
+	});
+
+	describe('lorsque le feature flip rend visible la page', () => {
+		beforeEach(() => {
+			process.env.NEXT_PUBLIC_OLD_ESPACE_JEUNE_FEATURE = '0';
+		});
+
+		it('appel le serveur pour récupérer les actualités', async () => {
+			jest.spyOn(dependencies.actualitesDependencies.consulterActualitesUseCase, 'handle').mockResolvedValue(createSuccess(anActualiteList()));
+
+			await getStaticProps();
+
+			expect(dependencies.actualitesDependencies.consulterActualitesUseCase.handle).toHaveBeenCalledTimes(1);
+		});
+
+		it('lorsque la récupération est en échec, redirige vers la page 404', async () => {
+			jest.spyOn(dependencies.actualitesDependencies.consulterActualitesUseCase, 'handle').mockResolvedValue(createFailure(ErreurMetier.SERVICE_INDISPONIBLE));
+
+			const result = await getStaticProps();
+
+			expect(result).toEqual({ notFound: true, revalidate: 1 });
+		});
```

Comment:
Quand on fait une disjonction de cas (par exemple : quand succès VS quand échec), j'aime bien faire deux blocs describe pour améliorer la lisibilité plutôt que de laisser un 'it' isolé et le reste en bloc. De cette façon, quand on lit juste les blocs au même niveau d'indentation on comprend logiquement qu'on couvre tous les cas possibles. C'est criant quand on utilise la fonctionnalité Collapse / Fold 

```suggestion
        describe('lorsque la récupération est en échec', () => {
		it('redirige vers la page 404', async () => {
			jest.spyOn(dependencies.actualitesDependencies.consulterActualitesUseCase, 'handle').mockResolvedValue(createFailure(ErreurMetier.SERVICE_INDISPONIBLE));

			const result = await getStaticProps();

			expect(result).toEqual({ notFound: true, revalidate: 1 });
		});
	});
```
On a le même cas pour le feature flip. Quand on est pas sur de la disjonction de cas mais juste tester plusieurs comportements isolé là je suis ok pour les 'it' isolés

Reply:
ce que je fais pour éviter la lourdeur des describe c'est que lorsqu'il y a plusieurs sous cas je fais un describe sinon je fais un it directement